Gems/Atom/Feature/Common/Assets/Materials/Types/EnhancedSurface_ForwardPass.azsl

```diff
@@ -0,0 +1,308 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+// SRGs
+#include <Atom/Features/PBR/ForwardPassSrg.azsli>
+
+// Pass Output
+#include <Atom/Features/PBR/ForwardSubsurfacePassOutput.azsli>
+
+// Utility
+#include <Atom/Features/ColorManagement/TransformColor.azsli>
+
+// Custom Surface & Lighting
+#include <Atom/Features/PBR/Lighting/EnhancedLighting.azsli>
+
+// Decals
+#include <Atom/Features/PBR/Decals.azsli>
+
+// ---------- Vertex Shader ----------
+
+struct VSInput
+{
+    // Base fields (required by the template azsli file)...
+    float3 m_position : POSITION;
+    float3 m_normal : NORMAL;
+    float4 m_tangent : TANGENT; 
+    float3 m_bitangent : BITANGENT; 
+ 
+    // Extended fields (only referenced in this azsl file)...
+    float2 m_uv0 : UV0;
+    float2 m_uv1 : UV1;
+};
+
+struct VSOutput
+{
+    // Base fields (required by the template azsli file)...
+    precise linear centroid float4 m_position : SV_Position;
+    float3 m_normal: NORMAL;
+    float3 m_tangent : TANGENT; 
+    float3 m_bitangent : BITANGENT; 
+    float3 m_worldPosition : UV0;
+    float3 m_shadowCoords[ViewSrg::MaxCascadeCount] : UV5;
+
+    // Extended fields (only referenced in this azsl file)...
+    float2 m_uv[UvSetCount] : UV1;    
```

Comment:
super nit: organize the UVs order and possibly add remark about using 2 UV set counts here (hence the 2 uv set spacing).

Reply:
No worries about being nit-picky :) but interpolants here were strictly moved as-is from the previous location in code. When a future change occurs to remove unused interpolants, we should clean this up.